analytics.go

```diff
@@ -418,12 +418,19 @@ func (c *client) send(msgs []message) {
 				We would then reset the node count by making a call to configure-info end point, then regenerate the payload at a node level
 				for only those nodes where we failed in sending the data and then recursively call the send function with the updated payload.
 				*/
-				c.logf("Retrying in 5 seconds")
-				time.Sleep(5 * time.Second)
-				c.setNodeCount()
-				newMsgs := c.getRevisedMsgs(nodePayload, k)
-				c.send(newMsgs)
-				return
+				var sleepTimeOut = time.Duration(count*5) * time.Second
+				if sleepTimeOut.Seconds() > 300 {
+					c.debugf("Discarding events")
```

Comment:
We don't want to discard the event, just keep retrying until it succeeds. Using a maximum value for the timeout is desirable though.

Reply:
@SumanthPuram mentioned keeping the maximum time to 5 minutes.

Reply:
we are guarding against a future bug in proxy. Under normal conditions the proxy won't keep returning a 451. So, 5 minutes is not enough to pinpoint and fix a production issue and will lead to data loss. If we keep the max backoff timeout to 5 minutes, that's ok, but we need to keep retrying

Reply:
So, what you suggest here?

Reply:
Got it.